cpp/include/cuspatial/experimental/point_quadtree.cuh

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/detail/utility/traits.hpp>
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/tuple.h>
+
+#include <tuple>
+
+namespace cuspatial {
+
+/**
+ * @addtogroup spatial_indexing
+ * @{
+ */
+
+struct point_quadtree {
+  // uint32_t vector of quad node keys
+  rmm::device_uvector<uint32_t> key;
+  // uint8_t vector of quadtree levels
+  rmm::device_uvector<uint8_t> level;
+  // bool vector indicating whether the node is a parent (true) or leaf (false) node
+  rmm::device_uvector<bool> is_parent_node;
+  // uint32_t vector for the number of child nodes (if is_parent_node), or number of points
+  rmm::device_uvector<uint32_t> length;
+  // uint32_t vector for the first child position (if is_parent_node), or first point position
+  rmm::device_uvector<uint32_t> offset;
+};
+
+/**
+ * @brief Construct a quadtree structure from points.
+ *
+ * @see http://www.adms-conf.org/2019-camera-ready/zhang_adms19.pdf for details.
+ *
+ * @note `scale` is applied to (x - min.x) and (y - min.y) to convert coordinates into a Morton code
```

Comment:
"applied" doesn't reflect that those differences are divided by scale...  If there is an intuitive explanation, let's add that. And if not, then put the exact math in here. See https://github.com/rapidsai/cuspatial/issues/397

Reply:
As discussed, should probably use a reciprocal of the current scale to make the scale factor actually a directly proportional scale factor (which would be more intuitive).

Reply:
Further discussion: let's save any semantic / breaking change for a later PR if necessary.